### ISSUE TITLE
Inconsistent behaviour between dev and build in parallel routing.

### ISSUE THREAD

**Issue by jainmohit2001** [vercel/next.js#59463]
### Link to the code that reproduces this issue

https://github.com/jainmohit2001/nextjs-features

### To Reproduce

1. Clone the mentioned project
2. `npm install`
3. `npm run dev` (for development)
4. `npm run build` followed by `npm run start` (for production server) 

### Current vs. Expected behavior

The [src/app/routing/parallel-routing/@slot1/layout.tsx](https://github.com/jainmohit2001/nextjs-features/blob/58e9256befc54d8de3394a4395af479fe24f104e/src/app/routing/parallel-routing/%40slot1/layout.tsx) should be visible for the `@slot1`. In dev mode, this works as expected, but in build mode, the layout seems to be missing. 

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP Wed Aug 25 23:20:18 UTC 2021
Binaries:
  Node: 20.10.0
  npm: 10.2.3
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.0.3
  eslint-config-next: 14.0.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Routing (next/router, next/navigation, next/link)

### Additional context

I tested the build with next canary release with the following info output:

```
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP Wed Aug 25 23:20:18 UTC 2021
Binaries:
  Node: 20.10.0
  npm: 10.2.3
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.0.5-canary.5
  eslint-config-next: 14.0.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.2
Next.js Config:
  output: N/A
```

**Comment by jainmohit2001** [comment]
So currently the parallel routes have a lot of bugs and there are still a lot of issues pending for the Next.js team to focus on, I went ahead and resolved this issue using `router.push()` and `router.refresh()` methods.

I know they will be solved soon, but releasing such big features without proper testing seems less professional. Considering I discovered this bug while toying with Next.js.

Anyhow, I'll keep this issue open until someone from the team presents a resolution!

**Comment by ztanner** [comment]
Hi there - thanks for the report. 

Slots do not support layout files -- that shouldn't be working in dev, so we'll take a look! Parallel routes are meant to be used within a shared layout.

Our [Parallel Routes docs](https://nextjs.org/docs/app/building-your-application/routing/parallel-routes) clarify this a bit more. The supported file types for a slot are `error`, `loading`, `page`, and `default`.

**Comment by jainmohit2001** [comment]
Hi @ztanner.

Thanks for the response! We would like to explicitly mention this in the docs as well.

Moreover, I am following the discussion https://github.com/vercel/next.js/discussions/54075. Please correct me if I am wrong:

As of now, there is no way to always load the latest data (in my case - a new random number at every request) when we are navigating back and forth between dynamic pages (`export const revalidate = 0`) due to [Router cache](https://nextjs.org/docs/app/building-your-application/caching#opting-out-3). We have to explicitly call the `router.refresh()` if we want to achieve this.

I am asking this because we are considering using Next.js as our framework for building new upcoming commercial products.
Our business requirements include NOT showing any stale data.

Thanks in advance!

**Comment by ztanner** [comment]
Hi @jainmohit2001 - yes indeed, I'll dig into the slot `layout` case a bit more, and we'll make sure to update the documentation based on the results from that.

Regarding caching, your understanding is correct. For dynamic segments, there's a 30s router cache. `router.refresh()` will purge the router cache and force a request to the server. There's a proposal for a `staletime` property (read more [here](https://github.com/vercel/next.js/discussions/54075#discussioncomment-6754339)) which will let you explicitly override the defaults to bypass the 30s heuristic.

**Comment by jainmohit2001** [comment]
Thanks a lot for your quick response @ztanner. 

The `staletime` proposal seems to address our requirements. Looking forward to that!
 
Closing this issue as `layout.tsx` is not supposed to be used with parallel route Slots.

**Comment by ztanner** [comment]
Hey @jainmohit2001 , just to follow-up, it turns out we did intend to support multiple layouts in parallel slots and I've fixed it in #61115

**Comment by jainmohit2001** [comment]
Hi @ztanner Thanks for the update!
